app/scripts/modules/core/src/pipeline/service/execution.service.ts

```diff
@@ -439,6 +454,27 @@ export class ExecutionService {
       }
     }
 
+    public hydrate(application: Application, unhydrated: IExecution): Promise<IExecution> {
+      if (unhydrated.hydrator) {
+        return unhydrated.hydrator;
+      }
+      const executionHydrator = this.getExecution(unhydrated.id)
+        .then(hydrated => {
+          this.transformExecution(application, hydrated);
+          hydrated.stages.forEach(s => {
+            const toHydrate = unhydrated.stages.find(s2 => s2.id === s.id);
+            if (toHydrate) {
+              toHydrate.context = s.context;
+              toHydrate.outputs = s.outputs;
+              toHydrate.tasks = s.tasks;
+            }
+          });
+          unhydrated.hydrated = true;
+          return unhydrated;
+        });
+      return Promise.resolve(executionHydrator);
```

Comment:
you *could* theoretically move this to an observable, but I'm not actually convinced it would be any better. If you were to change it, my guess is that the most sensible approach would be a `BehaviorSubject` that starts with a value of `null` and filters out `null` values to avoid calling subscribers until the request comes back, basically making it into a promise.

At that point you're basically just reimplementing what you have here, so my 2 cents is to keep it as is.